src/store/tests/Bridge/Azure/SearchStoreTest.php

```diff
@@ -0,0 +1,296 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\AI\Store\Tests\Bridge\Azure;
+
+use PHPUnit\Framework\Attributes\CoversClass;
+use PHPUnit\Framework\Attributes\Test;
+use PHPUnit\Framework\TestCase;
+use Symfony\AI\Platform\Vector\NullVector;
+use Symfony\AI\Platform\Vector\Vector;
+use Symfony\AI\Store\Bridge\Azure\SearchStore;
+use Symfony\AI\Store\Document\Metadata;
+use Symfony\AI\Store\Document\VectorDocument;
+use Symfony\Component\HttpClient\Exception\ClientException;
+use Symfony\Component\HttpClient\MockHttpClient;
+use Symfony\Component\HttpClient\Response\JsonMockResponse;
+use Symfony\Component\Uid\Uuid;
+
+#[CoversClass(SearchStore::class)]
+final class SearchStoreTest extends TestCase
+{
+    #[Test]
+    public function addDocumentsSuccessfully(): void
+    {
+        $httpClient = new MockHttpClient([
+            new JsonMockResponse([
+                'value' => [
+                    ['key' => 'doc1', 'status' => true, 'errorMessage' => null, 'statusCode' => 201],
+                ],
+            ], [
+                'http_code' => 200,
+            ]),
+        ]);
+
+        $store = new SearchStore(
+            $httpClient,
+            'https://test.search.windows.net',
+            'test-api-key',
+            'test-index',
+            '2023-11-01',
+        );
+
+        $uuid = Uuid::v4();
+        $document = new VectorDocument($uuid, new Vector([0.1, 0.2, 0.3]));
+
+        $store->add($document);
+
+        $this->assertSame(1, $httpClient->getRequestsCount());
```

Comment:
usually we use `self`  instead of `$this`, don't we?
```suggestion
        self::assertSame(1, $httpClient->getRequestsCount());
```

Reply:
oh, symfony/symfony is using `$this->`

Reply:
We should add a cs fixer rule

Reply:
see #202